pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -62,6 +63,8 @@ public abstract class AbstractReplicator {
 
     protected final String replicatorPrefix;
 
+    protected final AtomicBoolean isDisconnected = new AtomicBoolean(false);
```

Comment:
Can we just use bool type? `isDisconnected` is only used in synchronized methods.